lib/spack/spack/util/executable.py

```diff
@@ -144,7 +144,7 @@ def streamify(arg, mode):
 
         cmd = self.exe + list(args)
 
-        cmd_line = ' '.join(cmd)
+        cmd_line = "'%s'" % "' '".join(map(lambda arg: arg.replace("'", """'"'"'"""), cmd))
```

Comment:
There is the `shlex` module, but I think a command line with lots of escapes is harder to read than a quoted one (with the one exception of the single-quote-in-argument problem which is rare enough that I don't see any in spack currently). As for not using escape characters in the output, the idea is to try and keep the "add more escapes" problem down.

I'll fix the triple quoting; not very readable either way :) .